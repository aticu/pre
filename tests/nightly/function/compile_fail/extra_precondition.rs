use pre::pre;

#[pre("is bar")]
fn foo() {}

#[pre]
fn main() {
    #[assure("is bar", reason = "foo is bar")]
    #[assure("is baz", reason = "foo is baz")]
    foo()
}
