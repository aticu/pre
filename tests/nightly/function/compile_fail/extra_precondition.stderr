error[E0308]: mismatched types
 --> $DIR/extra_precondition.rs:8:5
  |
8 | /     #[assure("is bar", reason = "foo is bar")]
9 | |     #[assure("is baz", reason = "foo is baz")]
  | |______________________________________________^ expected a tuple with 1 element, found one with 2 elements
  |
  = note: expected struct `std::marker::PhantomData<(pre::CustomCondition<"is bar">,)>`
             found struct `std::marker::PhantomData<(pre::CustomCondition<"is bar">, pre::CustomCondition<"is baz">)>`
